sexp/src/main/java/com/novoda/sexp/SimpleEasyXmlParser.java

```diff
@@ -37,6 +70,92 @@ public static void parse(InputStream xml, Instigator instigator, XMLReader xmlRe
+    }
+
+    /**
+     * @param xml        the xml to parse
+     * @param instigator your fully created parser of the xml
+     * @param <T>        the expected type of your parsed XML
+     * @return your parsed XML as an object
+     */
+    public static <T> T parse(String xml, ElementFinderInstigator<T> instigator) {
+        return parse(new ByteArrayInputStream(xml.getBytes()), instigator, getDefaultSEXPXMLReader());
+    }
+
+    /**
+     * @param xml        the xml to parse
+     * @param instigator your fully created parser of the xml
+     * @param <T>        the expected type of your parsed XML
+     * @return your parsed XML as an object
+     */
+    public static <T> T parse(InputStream xml, ElementFinderInstigator<T> instigator) {
+        return parse(xml, instigator, getDefaultSEXPXMLReader());
+    }
+
+    /**
+     * @param xml        the xml to parse
+     * @param instigator your fully created parser of the xml
+     * @param xmlReader  is the interface that an XML parser's SAX2 driver must implement , using this? _bad ass_ alert
+     * @param <T>        the expected type of your parsed XML
+     * @return your parsed XML as an object
+     */
+    public static <T> T parse(String xml, ElementFinderInstigator<T> instigator, XMLReader xmlReader) {
+        return parse(new ByteArrayInputStream(xml.getBytes()), instigator, xmlReader);
+    }
+
+    /**
+     * @param xml        the xml to parse
+     * @param instigator your fully created parser of the xml
+     * @param xmlReader  is the interface that an XML parser's SAX2 driver must implement , using this? _bad ass_ alert
+     * @param <T>        the expected type of your parsed XML
+     * @return your parsed XML as an object
+     */
+    public static <T> T parse(InputStream xml, final ElementFinderInstigator<T> instigator, XMLReader xmlReader) {
+        final CountDownLatch latch = new CountDownLatch(1);
+        Instigator latchedInstigator = new Instigator() {
+            @Override
+            public RootTag getRootTag() {
+                return instigator.getRootTag();
+            }
+
+            @Override
+            public void create(RootElement rootElement) {
+                instigator.create(rootElement);
+            }
+
+            @Override
+            public void end() {
+                latch.countDown();
+            }
+        };
+
+        RootTag rootTag = instigator.getRootTag();
+        RootElement rootElement = new RootElement(rootTag.getNamespace(), rootTag.getTag());
+        rootElement.setEndElementListener(latchedInstigator);
+        instigator.create(rootElement);
+        xmlReader.setContentHandler(rootElement.getContentHandler());
+        XmlParser xmlParser = new XmlParser();
+        xmlParser.parse(xml, xmlReader);
+
+        try {
+            latch.await();
+        } catch (InterruptedException e) {
+            throw new RuntimeException();
```

Comment:
should be a specific exception, otherwise the client doesn't know what went wrong